src/NonBlocking/ConcurrentDictionary/DictionaryImplLong.cs

```diff
@@ -100,7 +100,7 @@ private bool TryClaimSlot(ref long entryKey, long key)
         {
             var entryKeyValue = entryKey;
             //zero keys are claimed via hash
-            if (entryKeyValue == 0 & key != 0)
+            if (entryKeyValue == 0 && key != 0)
```

Comment:
We expect unclaimed slot here so both conditions will be tested. Long time ago `&` helped by avoiding a branch.
Very likely it does not make any difference now.

Reply:
I assumed the same. However, surprisingly, [it still does](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunHgIIAGwYBZABQBLAHYYG2NA2mzgASgYBeAHxyN6hgAYGAMgbAGAQj36A3Fx53u9UwOEA5STLkKlptVp3qVsYmZpYGttQAvkA=).

Checking the generated asm in https://uica.uops.info/ sadly doesn't help because the tools it uses disagree. `llvm-mca` and `IACA 3.0` say the `&&` variant is slightly better even if it has `jne`, while `uiCA` says it is worse.

Reply:
Every CPU that matters nowdays speculates and has good enough branch prediction, so `&&` could as well be faster or just have more compact code. JIT will not try making this `&` as it is not an obvious improvement.
  
This `&&` could be a good candidate for a predicated compare on arm64, although I doubt JIT is using those.

Anyways. `&&` is more common style and the difference, if any, will be platform-specific and likely very small, so not worth chasing it.